opencti-platform/opencti-front/src/private/components/common/lists/FilterAutocomplete.tsx

```diff
@@ -0,0 +1,153 @@
+
+const FilterAutocomplete: FunctionComponent<FilterAutocompleteProps> = ({
+  filterKey,
+  defaultHandleAddFilter,
+  inputValues,
+  setInputValues,
+  availableEntityTypes,
+  availableRelationshipTypes,
+  allEntityTypes,
+  openOnFocus,
+}) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+
+  const [searchScope, setSearchScope] = useState<Record<string, Array<string>>>({});
+
+  const [entities, searchEntities]: [Record<string, Array<OptionValue>>, (filterKey: string, event: SyntheticEvent) => Record<string, Array<OptionValue>>] = useSearchEntities({
+    searchScope,
+    setInputValues,
+    availableEntityTypes,
+    availableRelationshipTypes,
+    allEntityTypes,
+  }) as [Record<string, Array<OptionValue>>, (filterKey: string, event: SyntheticEvent) => Record<string, Array<OptionValue>>]; // change when useSearchEntities will be in TS
+
+  const isStixObjectTypes: boolean = ['elementId', 'fromId', 'toId', 'objectContains'].includes(filterKey);
+
+  const handleChange = (event: SyntheticEvent, value: AutocompleteValue<OptionValue, boolean | undefined, boolean | undefined, boolean | undefined>) => {
+    if (value && !Array.isArray(value) && typeof value !== 'string') {
+      const group = !onlyGroupOrganization.includes(filterKey)
+        ? value.group
+        : undefined;
+      const filterAdd = `${filterKey}${group ? `_${group}` : ''}`;
+      defaultHandleAddFilter(filterAdd, value.value, value.label, event);
+    }
+  };
+
+  const renderSearchScopeSelection = (key: string) => (
+    <SearchScopeElement
+      name={key}
+      searchScope={searchScope}
+      setSearchScope={setSearchScope}
+    />
+  );
+
+  let options: Array<OptionValue> = [];
+  if (isStixObjectTypes) {
+    if (searchScope[filterKey] && searchScope[filterKey].length > 0) {
+      options = (entities[filterKey] || [])
+        .filter((n) => (searchScope[filterKey] || []).includes(n.type))
+        .sort((a, b) => (b.type ? -b.type.localeCompare(a.type) : 0));
+    } else {
+      options = (entities[filterKey] || []).sort((a, b) => (b.type ? -b.type.localeCompare(a.type) : 0));
+    }
+  } else if (entities[filterKey]) {
+    options = entities[filterKey];
+  }
+
+  return (
+    <Autocomplete
+      key={filterKey}
+      selectOnFocus={true}
+      openOnFocus={openOnFocus}
+      autoSelect={false}
+      autoHighlight={true}
+      getOptionLabel={(option) => (option.label ? option.label : '')}
+      noOptionsText={t('No available options')}
+      options={options}
+      onInputChange={(event) => searchEntities(filterKey, event)}
+      inputValue={inputValues[filterKey] as string || ''}
+      onChange={(event, value) => handleChange(event, value)}
+      groupBy={
+        isStixObjectTypes
+          ? (option) => option.type
+          : (option) => t(option.group ? option.group : `filter_${filterKey}`)
+      }
+      isOptionEqualToValue={(option, value) => option.value === value.value}
+      renderInput={(params) => (
```

Comment:
({ inputProps, ...params ]) => to prevent dissoc

Reply:
I'm not sure to understand the modification.
If I use this : renderInput={({ inputProps, ...params }) => (
The linter tell me that I'm not using inputProps later..

Maybe, I can just remove the dissoc because I use only the InputProps (with a capitale) property.

Reply:
It's fix by changing the assignement
The props InputProps is given by the Autocomplete field https://mui.com/material-ui/react-text-field/